pkg/tsdb/cloudwatch/credentials.go

```diff
@@ -28,7 +28,7 @@ var awsCredentialCache = make(map[string]cache)
 var credentialCacheLock sync.RWMutex
 
 func GetCredentials(dsInfo *DatasourceInfo) (*credentials.Credentials, error) {
-	cacheKey := dsInfo.AccessKey + ":" + dsInfo.Profile + ":" + dsInfo.AssumeRoleArn
+	cacheKey := fmt.Sprintf("%s:%s:%s:%s", dsInfo.AuthType, dsInfo.AccessKey, dsInfo.Profile, dsInfo.AssumeRoleArn)
```

Comment:
I can see that region is used below. What happens if someone changes the default region when the credentials are cached? Not sure if the current caching solution is preferred if multiple cloudwatch data sources using same credentials, but different default regions, and by that are able to reuse the same credentials?

One alternative to this would be to implement [same caching as used for data sources](https://github.com/grafana/grafana/blob/master/pkg/models/datasource_cache.go#L63) using datasource id as key and check if it have been updated and if so invalidate the cache. One advantage of this is that the cache will only be as large as the number of data sources where's the current approach is more dynamic and can in theory create a very large cache.